enhancements/user-workload-monitoring.md

```diff
@@ -0,0 +1,275 @@
+
+Instead, the "upgradable" condition in the ClusterOperator will be set to false.
+Routing configurations for alerts to be sent to specific teams need to be communicated/configured by an infrastructure owner using the existing Alertmanager stack that is part of cluster-monitoring.
+We will look into extension mechanisms post GA.
+With any OpenShift deployment, we only install the cluster-level part
+and allow infrastructure owner to "enable" the addition of user workload monitoring.
+
+### User Stories
+
+
+#### US1
+
+As a service owner, I’d like to configure a ServiceMonitor CRD so that Red Hat’s supported monitoring solution is able to scrape metrics for my service.
+
+#### US2
+
+As a service owner, I’d like to configure alerting rules for my service so that Red Hat’s supported monitoring solution notifies me if problems turn up.
+
+#### US3
+
+As a member of the operations team, I want to configure Prometheus one time so that all existing and any new namespaces will have access to user workload monitoring.
+
+#### US4
+
+As a member of the operations team, I'd like to add my own exporters so that I can monitor infrastructure services that are not monitored out-of-the-box.
+
+Note: This covers adding ServiceMonitor and PrometheusRule CRDs.
+
+#### US5
+
+As a developer working on a Red Hat product not part of the infrastructure, I’d like to integrate my metrics into a single, supported Prometheus stack so that they are available for my customers.
+
+#### US6
+
+As a service owner, I’d like to query the metrics I have access to from the OpenShift console.
+
+#### US7
+
+As a member of the operations tenant, I’d like to make sure that one tenant doesn't have access to metrics from another tenant.
+
+#### US8
+
+As a member of the operations team, I'd like the ability for application developers to access container cpu and memory usage metrics.
+
+### Implementation Details/Notes/Constraints
+
+#### Isolation
+
+The cluster monitoring as well as the user workload monitoring stack are composed of custom resources
+reconciled by two separate prometheus operators.
+Both prometheus operators are under supervision of cluster monitoring operator.
+This allows a separation of concerns in terms of upgradability, security settings, and scalability.
+The cluster monitoring Prometheus Operator continues to be deployed in the `openshift-monitoring` namespace.
+The user workload monitoring Prometheus Operator will be deployed in the `openshift-user-monitoring` namespace.
+
+Currently, cluster monitoring operator scans all namespaces filtering `openshift` namespaces labeled with
+`openshift.io/cluster-monitoring: "true"` and takes this set of namespaces as the _allowed list_ of reconcilable
+namespaces for the existing cluster monitoring prometheus operator.
+
+The same set of openshift namespaces that exist today will be configured as a _deny list_ of reconcilable namespaces
```

Comment:
implicit in this is that the user monitoring prometheus instance will be monitored by the cluster monitoring instance?

Reply:
yes :+1: The `openshift-user-monitoring` namespace is considered being an OpenShift namespace. ServiceMonitors placed in this namespace will be picked up by cluster monitoring. That way we will have a feedback loop via telemetry for the workload in that namespace, namely the prometheus operator as well as the new prometheus instances.

Reply:
Nice.